common/service/host/kcs.c

```diff
@@ -149,9 +150,44 @@ void kcs_read(void *arvg0, void *arvg1, void *arvg2)
 				memcpy(&bridge_msg.data[0], &ibuf[2], rc);
 			}
 
-			status = ipmb_send_request(&bridge_msg, IPMB_inf_index_map[BMC_IPMB]);
-			if (status != IPMB_ERROR_SUCCESS) {
-				LOG_ERR("kcs_read_task send to BMC fail status: 0x%x", status);
+
+			// Check BMC communication interface if use IPMB or not
+			if (!pal_is_interface_use_ipmb(IPMB_inf_index_map[BMC_IPMB])) {
+				int ret = 0;
+
+				// Send request to MCTP/PLDM thread to ask BMC
+				ret = pldm_send_ipmi_request(&bridge_msg);
+				if (ret < 0) {
+					LOG_ERR("kcs_read_task send to BMC fail");
+				}
+
+				uint8_t *kcs_buff;
+				kcs_buff = malloc(KCS_BUFF_SIZE * sizeof(uint8_t));
```

Comment:
Do we want to allocate a static buffer, or should we allocate 3+bridge_msg.data_len?

Reply:
Change to allocate 3 + bridge_msg.data_len, thank you.